pymc3/models/utils.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+from pandas.core.common import PandasError
+import numpy as np
+import theano.tensor as tt
+
+
+def any_to_tensor_and_labels(x, labels=None):
```

Comment:
It looks like splitting out `_input_to_array`, `_verify_labels`, then you can rewrite this as

```
x = _input_to_array(x)
labels = _verify_labels(labels)
assert_shape_labels(x.shape, labels)
return x, labels
```